torch/utils/_python_dispatch.py

```diff
@@ -66,12 +66,23 @@ def push(cls, *args, **kwargs):
 
 def _get_current_dispatch_mode():
     stack_len = _len_torch_dispatch_stack()
-    return _get_dispatch_stack_at(stack_len - 1) if stack_len > 0 else None
```

Comment:
Is there not a more canonical C++ function you can call to get this info, because this is the exact calculation the dispatcher needs to do? It would be best if we didn't have to duplicate the logic.

Reply:
agreed - this is basically exactly the `maybe_highest_mode()` util I added in this PR (which I'll refactor to its own PR). I can give it python bindings and use it here.